src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java

```diff
@@ -495,4 +578,16 @@ public String getByteLengthFunction(TypeRef typeRef) {
               "options",
               "callback")
           .build();
+
+  private static final ImmutableSet<String> COMMON_PROTO_PATHS =
```

Comment:
I do not have the answer to your question at this point. Let me investigate, but I'm afraid that MVVM will not be that great -- I mean, I suspect that this type of thing will be more specific to Node so either way logic like this would exist in somewhere specific to Node.